actix-session/src/cookie.rs

```diff
@@ -308,7 +308,7 @@ where
         Session::set_session(state.into_iter(), &mut req);
 
         Box::new(self.service.call(req).map(move |mut res| {
-            if let Some(state) = Session::get_changes(&mut res) {
+            if let (_status, Some(state)) = Session::get_changes(&mut res) {
```

Comment:
we should update cookies only if session has changes, otherwise we should not set cookie. i think this is the reason for #923

Reply:
the latest commit should reflect this